packages/client/hmi-client/src/views/TA2Playground.vue

```diff
@@ -527,6 +639,155 @@ export default defineComponent({
+			console.log(this.loadModelID);
+			const resp = await fetch(`http://localhost:8888/api/models/${this.loadModelID}/json`, {
+				method: 'GET'
+			});
+			const model = await resp.json();
+			this.createModel(model, false);
+		},
+		// Expects a JSON of a model with labels T, S, I, O.
+		// populates g + depending on provided flag POST changes to model ID
+		// This is mostly done for stratification testing. Will require a deeper look in future
+		// TODO: We know there are race errors here. We intend to make this service stateless so we wont need to add Edges and Nodes individually
+		async createModel(model, createFlag = false) {
+			// Flag is true so we need to call API PUT new model ID
+			if (createFlag === true) {
+				const newModel = await fetch('http://localhost:8888/api/models', { method: 'PUT' });
+				const modelData = await newModel.json();
+				modelId = modelData.id;
+				console.log(`Model ID: ${modelId}`); // currently required for testing
+			}
+
+			// Reset current nodes and edges
+			g.nodes = [];
+			g.edges = [];
+
+			const nodeHeight = 20;
+			const nodeWidth = 20;
+			let nodeX = 0;
+			let nodeY = 0;
+			// Nodes
+			for (let i = 0; i < model.S.length; i++) {
+				const aNode = model.S[i];
+				nodeX += 30;
+				nodeY += 30;
+				this.addNode(
+					`s-${i + 1}`,
+					aNode.sname.toString(),
+					nodeX,
+					nodeY,
+					nodeHeight,
+					nodeWidth,
+					NodeType.species,
+					createFlag
+				);
+			}
+			// Move Transitions 100 to the right of S
+			nodeX = 100;
+			nodeY = 0;
+			for (let i = 0; i < model.T.length; i++) {
+				const aTransition = model.T[i];
+				nodeX += 30;
+				nodeY += 30;
+				this.addNode(
+					`t-${i + 1}`,
+					aTransition.tname.toString(),
+					nodeX,
+					nodeY,
+					nodeHeight,
+					nodeWidth,
+					NodeType.transition,
+					createFlag
+				);
+			} // end T
+
+			// Edges
+			for (let i = 0; i < model.I.length; i++) {
+				const iEdges = model.I[i];
+				const sourceID = `s-${iEdges.is}`;
+				const transitionID = `t-${iEdges.it}`;
+				this.addEdgeID(sourceID, transitionID, createFlag);
+			}
+			for (let i = 0; i < model.O.length; i++) {
+				const oEdges = model.O[i];
+				const sourceID = `s-${oEdges.os}`;
+				const transitionID = `t-${oEdges.ot}`;
+				this.addEdgeID(transitionID, sourceID, createFlag);
+			}
+
+			// g = runLayout(_.cloneDeep(g));
+			this.refresh();
+			this.jsonOutput();
+		}, // end createModel
+		async stratify() {
+			try {
+				const outputModel = await fetchStratificationResult(
+					this.stratifyModelA,
+					this.stratifyModelB,
+					this.stratifyTypeModel
+				);
+				this.createModel(outputModel, true);
+			} catch (e: any) {
+				console.error(e.message);
+			}
+		},
+		// Used to create sample models for stratifying tests
+		// Will not be requried in the long run as we will be moving to storing these in DB
+		async createSampleModels() {
+			// TODO: Add Petri Net type to this when merged with other PR
+			const SIRDModel = {
```

Comment:
These can be typed correctly now that Shawn's PR is merged.

```suggestion
			const SIRDModel: PetriNet = {
```